src/topic_training/src/Demo.cpp

```diff
@@ -0,0 +1,136 @@
+/* Include all headers of the ROS system we are using 
+   separately. */
+#include "ros/init.h"
+#include "ros/node_handle.h"
+#include "ros/rate.h"
+#include "ros/subscriber.h"
+#include "ros/publisher.h"
+
+/* Include the message type we will be publishing. */
+#include "std_msgs/String.h"
+
+/* Include Stringstream to be able to manipulate the 
+   data we are sending and receiving. (Optional: only 
+   if the message is of type String)*/
+#include <sstream>
+
+/* C++ does not like "magic numbers," so it is better 
+   if we make them variables. */
+const int QUEUE_SIZE = 1000;
+const int FREQUENCY = 10;
+
+/**
+ * This function will get called when a new message has 
+ * arrived on the topic.
+ * 
+ * @param msg - the message received by the Subscriber, 
+ *              sent by the Publisher
+ */
+void chatterCallback(const std_msgs::String::ConstPtr& msg)
+{
+   /* Print out the message to the terminal. */
+   ROS_INFO("I heard: [%s]", msg->data.c_str());
+}
+
+/**
+ * The main method that contains the creation of the 
+ * node, Publisher, Subcriber, and topic. It also 
+ * describes what we will do with the data/message.
+ */
+auto main(int argc, char **argv) -> int
+{
+   /* Initialize ROS and specify the name of the node 
+      (which is the name of the file in this case). 
+      Requires the parameters from the main function 
+      header. */
+   ros::init(argc, argv, "Demo");
+
+   /* Create a handle for the node. This will do the 
+      actually initialization of the node in the next 
+      few lines. */
+   ros::NodeHandle nhandle;
+
+   /* Initialize the Publisher. The advertise() function 
+      is how we tell ROS we want to publish on a topic. 
+      State the message type we will be publishing 
+      (std_msgs::String), the name of the topic 
+      ("chatter"), and the queue size (1000). The queue 
+      size is the maximum number of messages kept as a 
+      buffer before throwing away old ones if we are 
+      publishing too quickly. */
+   ros::Publisher pub = nhandle.advertise<std_msgs::String>("chatter", QUEUE_SIZE);
+
+   /* Initialize the Subscriber. The subscriber() function 
+      is how we tell ROS we want to receive messages on 
+      a specific topic. State the name of the topic 
+      ("chatter"), the queue size (1000), and the function 
+      that will be called when a message is received 
+      (chatterCallBack(msg)). */
+   ros::Subscriber sub = nhandle.subscribe("chatter", QUEUE_SIZE, chatterCallback);
+
+   /* Specify the frequency we would like to loop at 
+      (10Hz). */
+   ros::Rate loop_rate(FREQUENCY);
+
+   /* A counter of how many messages have been sent. 
+      (Optional: only used for this example to show the 
+      difference in messages) */
+   int count = 0;
+   /* (Optional: a while loop is not required for
+      Publishers. Only used in this case to show the 
+      continuous timing of publishing messages.) */
+   while (ros::ok())
```

Comment:
Worth a note that loops waiting for 'actual time' should always be conditioned on `ros::ok`.

Reply:
Fixed on de76854.